pkg/sdn/plugin/sdn-cni-plugin/openshift-sdn.go

```diff
@@ -0,0 +1,463 @@
+		if err != nil {
+			return fmt.Errorf("Failed to fetch container veth: %v", err)
+		}
+
+		// Clear out gateway to prevent ConfigureIface from adding the cluster
+		// subnet via the gateway
+		result.IP4.Gateway = nil
+		if err = ipam.ConfigureIface(interfaceName, result); err != nil {
+			return fmt.Errorf("Failed to configure container IPAM: %v", err)
+		}
+
+		lo, err := netlink.LinkByName("lo")
+		if err == nil {
+			err = netlink.LinkSetUp(lo)
+		}
+		if err != nil {
+			return fmt.Errorf("Failed to configure container loopback: %v", err)
+		}
+		return nil
+	})
+	if err != nil {
+		ipam.ExecDel("host-local", ipamConfig)
+		return err
+	}
+
+	if macvlan {
+		if err := addMacvlan(args.Netns); err != nil {
+			ipam.ExecDel("host-local", ipamConfig)
+			return err
+		}
+	}
+
+	contVethMac := contVeth.Attrs().HardwareAddr.String()
+	podIP := result.IP4.IP.IP.String()
+	out, err := exec.Command(sdnScript, setUpCmd, hostVeth.Attrs().Name, contVethMac, podIP, vnid, ingress, egress).CombinedOutput()
+	if isScriptError(err) {
+		ipam.ExecDel("host-local", ipamConfig)
+		return fmt.Errorf("Error running network setup script:\nhostVethName %s, contVethMac %s, podIP %s, vnid %s, ingress %s, egress %s\n %s", hostVeth.Attrs().Name, contVethMac, podIP, vnid, ingress, egress, out)
+	} else if err != nil {
+		ipam.ExecDel("host-local", ipamConfig)
+		return err
+	}
+
+	return result.Print()
+}
+
+func cmdUpdate(args *skel.CmdArgs) error {
+	_, vnid, ingress, egress, _, err := getPodInfo(args)
+	if err != nil {
+		return err
+	}
+
+	hostVeth, contVethMac, podIP, err := getVethInfo(args.Netns, args.IfName)
+	if err != nil {
+		return err
+	}
+
+	out, err := exec.Command(sdnScript, updateCmd, hostVeth.Attrs().Name, contVethMac, podIP, vnid, ingress, egress).CombinedOutput()
+
+	if isScriptError(err) {
+		return fmt.Errorf("Error running network update script: %s", getScriptError(out))
+	} else if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func cmdDel(args *skel.CmdArgs) error {
+	nodeConfig, _, _, _, _, err := getPodInfo(args)
+	if err != nil {
+		return err
+	}
+
+	hostVeth, contVethMac, podIP, err := getVethInfo(args.Netns, args.IfName)
+	if err != nil {
+		return err
+	}
+
+	// The script's teardown functionality doesn't need the VNID
+	out, err := exec.Command(sdnScript, tearDownCmd, hostVeth.Attrs().Name, contVethMac, podIP, "-1").CombinedOutput()
+
+	if isScriptError(err) {
+		return fmt.Errorf("Error running network teardown script: %s", getScriptError(out))
+	} else if err != nil {
+		return err
+	}
+
+	// Run IPAM to release the IP address lease
+	os.Setenv("CNI_ARGS", "")
+	if err := ipam.ExecDel("host-local", getIPAMConfig(nodeConfig)); err != nil {
+		return fmt.Errorf("Failed to run CNI IPAM DEL: %v", err)
+	}
+
+	return nil
+}
+
+func main() {
+	addFunc := cmdAdd
+
+	// CNI doesn't yet have an UPDATE command, so fake it
+	cmd := os.Getenv("CNI_COMMAND")
+	if cmd == "UPDATE" {
+		addFunc = cmdUpdate
+		os.Setenv("CNI_COMMAND", "ADD")
```

Comment:
Yes, https://github.com/containernetworking/cni/issues/89.  I'll update the comment to reflect that.